src/test/statistics/chunk_statistics/counting_quotient_filter_test.cpp

```diff
@@ -91,14 +91,43 @@ class CountingQuotientFilterTest : public BaseTest {
   }
 
   template <typename DataType>
-  void test_can_prune(std::shared_ptr<CountingQuotientFilter<DataType>> cqf, std::map<DataType, size_t>
+  void test_can_not_prune(std::shared_ptr<CountingQuotientFilter<DataType>> cqf, std::map<DataType, size_t>
       value_counts) {
     for (auto value_and_count : value_counts) {
         EXPECT_FALSE(cqf->can_prune(value_and_count.first, PredicateCondition::Equals));
     }
   }
+
+  template <typename DataType>
+  DataType get_test_value(int run) {
+    throw std::logic_error("specialize this method");
+  }
+
+  template <typename DataType>
+  void test_false_positive_rate(std::shared_ptr<CountingQuotientFilter<DataType>> cqf) {
+    int runs = 1000;
```

Comment:
Please use uint32_t or similar. I like and don't like this test. Even though it's not pretty, I would still consider it very useful.
Could you please add a short "warning" that this test is somewhat probabilistic?

Reply:
What does probabilistic mean here? Can the test fail because it has a bad day?

Reply:
Ah, saw the discussion above. Where does the randomness come from?

Reply:
When asking whether a value is contained in a segment there is a false positive rate. This means that if we want to test whether values that are not contained can be pruned, there will be a false positive rate aswell. I added a test that does 1000 queries for uncontained values and calculates the false positive rate. The test passes if the false positive rate is under 10%.